lib/wasix/src/net/socket.rs

```diff
@@ -72,8 +72,28 @@ pub enum InodeSocketKind {
         socket: Box<dyn VirtualUdpSocket + Sync>,
         peer: Option<SocketAddr>,
     },
-    RemoteTcpStream {
+    RemoteSocket {
+        family: Addressfamily,
+        ty: Socktype,
+        pt: SockProto,
```

Comment:
This is getting really complex, should probably live in a separate struct.

Reply:
done